### PR TITLE
UPDATED Changes to list options edit.

### PR DESCRIPTION
**Pull request by bradymiller**
hi @yurige-matrix ,
Here's an update from your prior PR. I made no changes; had to make the new because of a merge conflict and the building was broken (not your fault; npm had a vulnerability that was fixed a week ago that disrupted code before that). Your prior PR was:
https://github.com/openemr/openemr/pull/1988

### CODE REVIEW

**Comment by bradymiller** [comment]
here's @yurige-matrix 's description of this feature:
"
Hi, i'm just add a feature to update the list options without deleting all items.
There is a new functionality to limit a number of result that you get in each list.
We did is because, when there are many rows in the list browser crashes.
Waiting for your comments,
"

**Comment by bradymiller** [comment]
hi @yurige-matrix ,

This isn't working for me on testing.

Here's what I see when go to the Lists page:
image

And then if I click 'Show Records', it shows the following:
image

**Comment by yurige-matrix** [comment]
I committed to the other branch, do you want me to checkout and commit to this new branch? or you can merge from there the fix i did?

**Comment by bradymiller** [comment]
hi @yurige-matrix , I just brought in your fix into this PR. -brady

**Comment by bradymiller** [comment]
hi @yurige-matrix ,
On testing, I can not add entries. If I add the information entries and click Save, they are not added. Also rec. defaulting the top range to show everything (maybe leaving it blank would mean to show everything, which would do by default).
-brady

**Comment by bradymiller** [comment]
hi @yurige-matrix ,
Thanks for the reminder. Just brought in the fix and will test/review this by tomorrow.
-brady

**Comment by bradymiller** [comment]
hi @yurige-matrix ,
Sorry didn't get to this tonight. Will be the first thing I do tomorrow.
-brady

File: interface/super/edit_list.php [hunk 1]
```diff
@@ -43,6 +43,11 @@
     die(xlt('Not authorized'));
 }
 
+//Limit variables for filter
+$list_from = ( isset($_GET["list_from"]) ? intval($_GET["list_from"]) : (isset($_POST["list_from"]) ? intval($_POST["list_from"]) : 0) );
+$list_to   = ( isset($_GET["list_to"]) ? intval($_GET["list_to"]) : (isset($_POST["list_to"]) ? intval($_POST["list_to"]) : 10) );
```
Comment (bradymiller) [inline interface/super/edit_list.php]: Couldn't this be simplified if use $_REQUEST ?

File: interface/super/edit_list.php [hunk 2]
```diff
@@ -1064,6 +1095,46 @@ function mysubmit() {
...
+                        if( list_id_container.length > 0 ){
+                            urlFull.searchParams.delete("list_id_container");
+                            urlFull.searchParams.set("list_id_container", list_id_container );
+                        }
+                        window.location.replace(urlFull);
+                    }
+                </script>
+                <?php
+                    $urlFrom = ( isset($_GET["list_from"]) && intval($_GET["list_from"]) >= 0 ? intval($_GET["list_from"]) : (isset($_POST["list_from"]) && intval($_POST["list_from"]) >= 0 ? intval($_POST["list_from"]) : 0) );
+                    $urlTo   = ( isset($_GET["list_to"]) && intval($_GET["list_to"])  >= 0 ? intval($_GET["list_to"]) : (isset($_POST["list_to"]) && intval($_POST["list_to"]) >= 0 ? intval($_POST["list_to"]) : 10) );
```
Comment (bradymiller) [inline interface/super/edit_list.php]: Couldn't this be simplified if use $_REQUEST ?

**Comment by bradymiller** [comment]
Seems to work on light testing (will plan more heavy testing soon).

The flow is a bit confusing, which I think can be alleviated by not setting a upper limit by default; then a user will only use it when they need it (and a user won't mistakenly assume there are only 10 values in a list).
1. Make the To entry blank by default.
2. When the To entry is set to blank, then use no upper limit

Also, the from entry starting from 0 seems weird. Shouldn't this be 1 since users generally don't think in terms of arrays like we do :) ?

File: interface/super/edit_list.php (hunk 1 above)
Comment (yurige-matrix) [reply interface/super/edit_list.php]: You are right.

File: interface/super/edit_list.php (hunk 2 above)
Comment (yurige-matrix) [reply interface/super/edit_list.php]: You are right.

**Comment by yurige-matrix** [comment]
> Make the To entry blank by default.
When the To entry is set to blank, then use no upper limit

Hi @bradymiller , just fixed this. But i think, may be better make a setting for lists: "List options page size: " and default value 40. Because if somebody have largest lists, he'll get freeze on page in browser, when he want see this list. 
What you think about this?

**Comment by bradymiller** [comment]
hi @yurige-matrix ,
I'll do some testing (over next 1-2 days) with your recent changes and get back to you.
thanks,
-brady

**Comment by yurige-matrix** [comment]
hi @bradymiller , When can you do the testing?

File: interface/super/edit_list.php [hunk 3]
```diff
@@ -1023,6 +1045,14 @@ function mysubmit() {
                     <select name='list_id' class="form-control select-dropdown"
                             id="list_id">
                         <?php
+                        /*
+                         * Keep proper list name (otherwise list name changes according to
+                         * the options shown on the screen).
+                         */
+                        $list_id_container = trim(strip_tags($_GET["list_id_container"]));
```
Comment (bradymiller) [inline interface/super/edit_list.php]: What is the strip_tags() function call meant to do?

Comment (bradymiller) [inline interface/super/edit_list.php]: hi @yurige-matrix , What is the strip_tags() call meant to prevent?

File: interface/super/edit_list.php [hunk 4]
```diff
@@ -1064,6 +1094,49 @@ function mysubmit() {
...
+                        if( list_from > list_to ){
+                            alert(<?php echo xlj("Please enter a enter valid range"); ?>);
+                            return false;
+                        }
+                        if( parseInt(list_from) >= 0 ){
+                            urlFull.searchParams.delete("list_from");
+                            urlFull.searchParams.set("list_from",list_from);
+                        }
+
+                        if( parseInt(list_from) >= 0 ){
```
Comment (bradymiller) [inline interface/super/edit_list.php]: hi @yurige-matrix , This if and the of above are the same. Is they meant to be the same?

File: interface/super/edit_list.php (hunk 3 above)
Comment (yurige-matrix) [reply interface/super/edit_list.php]: hi, @bradymiller , prevent html in url,

File: interface/super/edit_list.php (hunk 4 above)
Comment (yurige-matrix) [reply interface/super/edit_list.php]: Fixed

**Comment by yurige-matrix** [comment]
Hi @bradymiller.
You are have made a review of code ?

**Comment by bradymiller** [comment]
thanks for reminder! Will place at top of my queue (likely can't get to this until the weekend, though)

**Comment by yurige-matrix** [comment]
Hi @bradymiller .Much time passed waiting for viewing the code. We need to implement this functionality in the system. If this is possible on your part, we will be happy, and if not, we will look for new solutions.

**Comment by sjpadgett** [comment]
@yurige-matrix I loaded this to test but my first thought was that while this may solve loading long lists, it doesn't seem practical from an editing standpoint. At minimum it should have a search resource and the from/to inputs should track. eg. From: 40 To: 80 of 'total count of list' then after relist From: 80 To: 'next To value' etc. or set a max list displayed value and have a Jump To input.
May be best to add pagination alphabetically or numerically. Otherwise, one is guessing where in list they need to go to edit a list item.

**Comment by yurige-matrix** [comment]
Hi @sjpadgett, I think that in this list it will be the simplest solution for a split of the list. You can split the list into pages, but this will only confuse the user.

**Comment by sjpadgett** [comment]
> confuse the user

I suppose not any more confusing than having a list of 500 items where you want to edit an item in the list. How do you find that item? Not sure how large a list this PR is addressing but it'd have to fairly long to exhibit locking the browser. 
All i'm saying is that if we have to manage long lists in the UI then we should have some tools to do so. At minimum could we at least hide the from/to inputs if list is short and/or at least display the total number of items in the list.
This type of long list may not come up very often(first i've heard of this) so I may be over critical of UI issues but i'd really like to see controls hidden for short lists.
I won't hold PR up any longer for my concerns and up to @bradymiller if this PR past muster just would appreciate if my minimum concerns are address. Also i've tested as far as adding and deleting from list and don't see any issues there. 
Okay, i'm off to bug someone else ;) Thanks

**Comment by yurige-matrix** [comment]
Hi @sjpadgett. Let's wait what Brady will says.

**Comment by bradymiller** [comment]
Was having some merge conflicts, so rebased to most recent codebase and created next PR here: https://github.com/openemr/openemr/pull/2213
(will close this PR, but will continue discussion started here by @sjpadgett there after I do some testing)

**Pull request closed without merging**